Improve cloudserver dashboards

francoisferrand opened the pull request:
* Add overview panels (number of buckets, number of objects, number of Up replicas)
* Add missing "Top10 buckets with errors" panels
* Add request duration heatmap
* Add upload/download chunk size distribution
* Fix unit for http responses

Issue: CLDSRV-74

<img width="1676" alt="Screenshot 2021-12-29 at 16 24 06" src="https://user-images.githubusercontent.com/3909027/147677922-c4e11f2a-990c-4d13-97fc-c6e12dc66033.png">

francoisferrand committed:
Add overview panels on top

- "up" replicas counter
- "Total requests" counter
- Number of buckets & objects
- Data disk storage
- Data & object injection rate
- Responses counter by category (200, 4xx, 5xx)

Issue: CLDSRV-74

francoisferrand committed:
Improve response code graphs

- Compute the "increase" instead of "rate", to display the actual number
of operations
- Use "multi" tooltips, to allow viewing the value under cursor
- Use separate colors and stack series for aggregated statuses
- Fill graphs & use smooth lines

Issue: CLDSRV-74

francoisferrand committed:
Improve latency display

- Display overall and per-action latencies on the same graph
- Add heatmap showing the request duration "repartition" over time

Issue: CLDSRV-74

francoisferrand committed:
Improve data rate display

- Use non-SI byte unit (kB/s, MB/s...)
- Fix capitalization of label
- Display upload packet size distribution
- Display download packet size distribution

Issue: CLDSRV-74

francoisferrand committed:
Display top 10 buckets with errors

Separate panels for 404, 500 and 50X errors.

Issue: CLDSRV-74

bert-e commented:
# Hello francoisferrand,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/cloudserver/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue CLDSRV-74 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `8.3.6`


Please check the `Fix Version/s` of CLDSRV-74, or the target
branch of this pull request.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

williamlardier commented on monitoring/dashboard.json:
```diff
@@ -16,35 +24,1214 @@
+            "type": "constant",
+            "label": "job",
+            "description": "Name of the Cloudserver job, used to filter only the Cloudserver instances.",
+            "value": "artesca-data-connector-s3api-metrics"
```
Just out of curiosity, is it a good thing to hardcode the artesca prefix here? How is the testing of this feature within other CIs if the prefix is different?

francoisferrand replied:
it is not really "hard-coded" : in production, the value is actually computed by Zenko-operator, and this value is actually not used.

However, having the value here makes it easier to load (manually) the dashboard in an artesca setup : grafana will prompt to get the values for the __inputs, but it will pre-populate with the values specified here...

williamlardier replied:
I see, thanks! :)

williamlardier reviewed (commented):
A quick comment before approving :slightly_smiling_face: 
Also, why does the latency on your bottom right screen go from 500 ms to +Inf? Is this on purpose? Maybe a better display would make more sense, replacing for example +Inf by 600ms.

francoisferrand commented:
> why does the latency on your bottom right screen go from 500 ms to +Inf? Is this on purpose? Maybe a better display would make more sense, replacing for example +Inf by 600ms.

There is a known bug in the metrics : the values are completely wrong, by a factor of 1000... So normally we would have much smaller values.

+Inf is the upper value of the range of the last bucket... I am not sure I can have much impact on how/what grafana will display in that case :-/ (though hopefully it will look better with the metrics fix, as it does in vault dashboard)

lavigne958 commented on monitoring/dashboard.json:
```diff
@@ -16,35 +24,1214 @@
+            "targets": [
+                {
+                    "exemplar": true,
+                    "expr": "sum(http_requests_total{namespace=\"${namespace}\", job=\"${job}\"})",
```
I disagree with this formula. I see no points watching a counter growing.
If we want to measure how many requests we are currently receiving we must use something like:
`sum(rate(http_requests_total{namespace="${namespace}", job="${job}"}[1h]))`

francoisferrand replied:
yes and no :-)

This panel is a **stat**, showing the "Total requests" : soi the value we want to display is the current one.
In addition, the value computed with `rate` (or actually, `increase`) is not correct  : seems to be some rounding error, and we don't reach the same value...

lavigne958 replied:
I understand and you currently display the latest recorded value for that counter, but as life itself will go on, the counter will only increase, so what we read here is the number of served requests since the platform has been setup. this is nice for breaking records but I don't see the use of it in monitoring :open_mouth: I don't mind keeping it though

lavigne958 commented on monitoring/dashboard.json:
```diff
@@ -16,35 +24,1214 @@
+            "targets": [
+                {
+                    "exemplar": true,
+                    "expr": "sum(cloud_server_number_of_buckets{namespace=\"${namespace}\",job=\"${job}\"})",
```
I currently don't have a running platform but, this is a gauge right ?

francoisferrand replied:
yes

lavigne958 commented on monitoring/dashboard.json:
```diff
@@ -16,35 +24,1214 @@
+            "targets": [
+                {
+                    "exemplar": true,
+                    "expr": "sum(cloud_server_number_of_objects{namespace=\"${namespace}\",job=\"${job}\"})",
```
same here this is a gauge right ?

francoisferrand replied:
yes

lavigne958 commented on monitoring/dashboard.json:
```diff
@@ -16,35 +24,1214 @@
+            "targets": [
+                {
+                    "exemplar": true,
+                    "expr": "sum(http_requests_total{namespace=\"${namespace}\",job=\"${job}\",code=\"200\"})",
```
same as `http_requests_total` their is no point displaying the counter itself.

lavigne958 commented on monitoring/dashboard.json:
```diff
@@ -16,35 +24,1214 @@
+            "targets": [
+                {
+                    "exemplar": true,
+                    "expr": "sum(http_requests_total{namespace=\"${namespace}\", job=\"${job}\",code=~\"4..\"})",
```
same here, no need to display a pure counter.
I know I am annoying :-) with this

lavigne958 commented on monitoring/dashboard.json:
```diff
@@ -16,35 +24,1214 @@
+            "targets": [
+                {
+                    "exemplar": true,
+                    "expr": "sum(http_requests_total{namespace=\"${namespace}\", job=\"${job}\",code=~\"5..\"})",
```
same here, no need to display a pure counter without a formula.

lavigne958 commented on monitoring/dashboard.json:
```diff
@@ -16,35 +24,1214 @@
+            "targets": [
+                {
+                    "exemplar": true,
+                    "expr": "sum(http_active_requests{namespace=\"${namespace}\", job=\"${job}\"})",
+                    "instant": false,
```
This should have the value `instant=true`, because we only care about the last non null value only.

francoisferrand replied:
not really : since we have the "background" graph : showing the history of this gauge

lavigne958 commented on monitoring/dashboard.json:
```diff
@@ -16,35 +24,1214 @@
+            "targets": [
+                {
+                    "exemplar": true,
+                    "expr": "sum(up{namespace=\"${namespace}\", job=\"${job}\"})",
+                    "instant": false,
```
same here I believe the `instant=true` should be set, we only want the last non null value.

francoisferrand replied:
not really : since we have the "background" graph : showing the history of this gauge

francoisferrand committed:
Display the "current" http requests count stat

Used to display the "total" count, show only the increase over current
time range.

Issue: CLDSRV-74

francoisferrand commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/8.3`



The following branches will **NOT** be impacted:

* `development/6.4`
* `development/7.10`
* `development/7.4`
* `development/8.2`
* `stabilization/7.4.10`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/8.3`



The following branches have **NOT** changed:

* `development/6.4`
* `development/7.10`
* `development/7.4`
* `development/8.2`
* `stabilization/7.4.10`



Please check the status of the associated issue CLDSRV-74.

Goodbye francoisferrand.
